hail/python/test/hail/linalg/test_linalg.py

```diff
@@ -991,3 +1017,56 @@ def test_filtering(self):
         with pytest.raises(ValueError) as exc:
             bm.filter_rows([0]).filter_rows([3]).to_numpy()
         assert "index" in str(exc)
+
+
+    @skip_unless_spark_backend()
+    def test_sparsify_blocks(self):
+        block_list = [1, 2]
+        np_square = np.arange(16, dtype=np.float64).reshape((4, 4))
+        block_size = 2
+        bm = BlockMatrix.from_numpy(np_square, block_size=block_size)
+        bm = bm._sparsify_blocks(block_list)
+        sparse_numpy = sparsify_numpy(np_square, block_size, block_list)
+        assert np.array_equal(bm.to_numpy(), sparse_numpy)
```

Comment:
I think it's worth adding this line for this small case as it aids others who come to these tests trying to understand what's happening:
```python3
assert np.array_equal(
    sparse_numpy,
    np.array([[0,  0,  3, 4],
              [0,  0,  7, 8],
              [9,  10, 0, 0],
              [13, 14, 0, 0]]))
```